firmware/src/src/targets/nrf52/ble.cpp

```diff
@@ -100,7 +108,7 @@ void BTSetMode(btmodet mode)
   // Requested same mode, just return
   if (mode == curmode) return;
 
-  btThreadRun = false;
+  k_sem_give(&btPauseSem);
```

Comment:
what did this help with -- i see a few around the 'pause' - hmm

Reply:
Actually not much. I was having some other issues around the threads and wondered if it was maybe the startup sequence so I switched them all to thread safe ones over the volatile bools. Just didn't get around to converting the pause for flash. Should really be tested again if its actually needed if it is should be switched to a proper method.